contracts/staking/StakeManager.sol

```diff
@@ -165,6 +176,40 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
     emit ReStaked(validatorId, validators[validatorId].amount, totalStaked);
   }
 
+  function claimRewards(uint256 validatorId, uint256 accountBalance, uint256 index, bytes memory proof) public /*onlyStaker(validatorId) */ {
+    // accountState = keccak256(abi.encodePacked(validatorId, accountBalance))
+    require(keccak256(abi.encodePacked(validatorId, accountBalance)).checkMembership(index, accountStateRoot, proof));
+    uint256 _reward = accountBalance.sub(validators[validatorId].totalReward);
+    address _contract = validators[validatorId].contractAddress;
+    if (_contract == address(0x0)) {
+      validators[validatorId].reward = validators[validatorId].reward.add(_reward);
+    }
+    else {
+      // TODO: delegator bond/share rate if return needs to be updated periodically
+      // otherwise validator can delay and get all the delegators reward
+      ValidatorContract(_contract).updateRewards(_reward, currentEpoch, validators[validatorId].amount);
+    }
+    totalRewardsLiquidated += _reward;
+    require(totalRewardsLiquidated <= totalRewards, "Oops this shouldn't have happened");// pos 2/3+1 is colluded
+    validators[validatorId].totalReward = accountBalance;
+    emit ClaimRewards(validatorId, _reward, accountBalance);
+  }
+
+  function withdrawRewards(uint256 validatorId) public onlyStaker(validatorId) {
+    uint256 amount = validators[validatorId].reward;
+    address _contract = validators[validatorId].contractAddress;
+    if (_contract != address(0x0)) {
+      amount = amount.add(ValidatorContract(_contract).withdrawRewardsValidator());
+    }
+    validators[validatorId].reward = 0;
+    require(token.transfer(msg.sender, amount), "Insufficent rewards");
+  }
+
+  function delegationTransfer(uint256 amount, address delegator) public {
```

Comment:
Where is this being used?

Reply:
ah! forgot to use it in delegation manager.

Reply:
Thanks